### PR TITLE
[BREAKING] Transform module to ESM

### PR DESCRIPTION
**Pull request by RandomByte**
See also https://github.com/SAP/ui5-logger/pull/306

### CODE REVIEW

**Comment by RandomByte** [comment]
Rebased onto latest `next`

**Comment by RandomByte** [comment]
LGTM

**Review by flovogt — CHANGES_REQUESTED** [review]

File: test/bin/ui5.js [hunk 1]
```diff
@@ -1,150 +1,252 @@
...
-		t.deepEqual(consoleLogStub.getCall(3).args,
-			[""]);
-		t.deepEqual(consoleLogStub.getCall(4).args,
-			["=> Please upgrade to a supported version of Node.js to use this tool"]);
-		t.deepEqual(consoleLogStub.getCall(5).args,
-			["====================================================================="]);
-
-		t.end();
+	// esmock.p is needed as import-local is loaded via dynamic import
+	t.context.ui5 = await esmock.p("../../bin/ui5.js", {
```
[the remark this answers is not in the document]
Comment (flovogt) [reply test/bin/ui5.js:74]: Done

File: lib/cli/cli.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		pkg,
+		updateCheckInterval: 86400000, // 1 day
+		shouldNotifyInNpmScript: true
+	}).notify();
+	// Remove --no-update-notifier from argv as it's not known to yargs, but we still want to support using it
+	const NO_UPDATE_NOTIFIER = "--no-update-notifier";
+	if (process.argv.includes(NO_UPDATE_NOTIFIER)) {
+		process.argv = process.argv.filter((v) => v !== NO_UPDATE_NOTIFIER);
+	}
+
```
[the remark this answers is not in the document]
Comment (flovogt) [reply lib/cli/cli.js:26]: Done

File: lib/cli/cli.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		.map((e) => new URL(`./commands/${e.name}`, import.meta.url));
+}
+
+export default async (pkg) => {
+	updateNotifier({
+		pkg,
+		updateCheckInterval: 86400000, // 1 day
+		shouldNotifyInNpmScript: true
+	}).notify();
+	// Remove --no-update-notifier from argv as it's not known to yargs, but we still want to support using it
```
[the remark this answers is not in the document]
Comment (flovogt) [reply lib/cli/cli.js:21]: Done

File: test/lib/cli/cli.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+			isDirectory: sinon.stub().returns(false),
+			name: ".DS_Store"
+		}
+	]);
+
+	t.context.cli = await esmock.p("../../../lib/cli/cli.js", {
+		"update-notifier": t.context.updateNotifier,
+		"yargs": t.context.yargs,
+		// TODO: Somehow esmock is unable to mock this import
+		// "yargs/helpers": {
```
[the remark this answers is not in the document]
Comment (flovogt) [reply test/lib/cli/cli.js:56]: @matz3 has coded a workaround for this issue. can be ignored for now.

**Review by flovogt — APPROVED** [review]

**Pull request merged**